Numerous changes

dstolfa opened the pull request:
This pull request elaborates on the branching instructions, updates pseudo-code to reflect reality more and explains some of the instructions better.

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -16,7 +16,10 @@ \section{The DIF Interpreter}
+until they are exhausted or an error causes interpretation to end. The DIF
+objects are verified in the \subroutine{dtrace_difo_validate} subroutine and
+the DIF interpreter ignores any bounds checking within the
+\subroutine{dtrace_dif_emulate} subroutine for precisely this reason.
```
"precisely this reason"  I think this needs to be reworded or you need to explain the reason.

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -56,14 +59,22 @@ \subsection{Comparison and Test Instructions}
+\subsection{Branching Instructions}
+
+DIF has eleven branching instructions split into two types: signed and unsigned
+branching instructions. The signed branching instructions take into account
```
s/branching instructions//

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -56,14 +59,22 @@ \subsection{Comparison and Test Instructions}
+that the number may be negative, while the unsigned instructions are meant to
+be used with exclusively positive numbers. One thing all of the branching
+instructions have in common is that they load the new \register{pc} register
+from the \registerop{label} field in the Branch Format as will described in
```
as described in (no need to say 'will be')

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -129,8 +140,14 @@ \subsection{Global Variables}
+references to simple names within the D script.  Space for global
+variables is statically allocated on each invocation of a script. Additionally,
+global variables are identified using the modified register format as
+will be describ
```
s/will be//

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -129,8 +140,14 @@ \subsection{Global Variables}
+global variables are identified using the modified register format as
+will be describ
+ed in Subsection~\ref{subsec:r-format} the case of arrays and the
+wide-immediate format which will be described in
```
s/will be//

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -177,7 +194,34 @@ \subsection{Register Format (R-Format)}
 \end{description}
 
+
+Furthermore, a modified version of the Register Format is used when loading and
```
s/Futhermore//

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -177,7 +194,34 @@ \subsection{Register Format (R-Format)}
 
+
+Furthermore, a modified version of the Register Format is used when loading and
+storing to array variables in DTrace. The main difference between the regular
```
s/to/data in/

gvnn3 commented on specification/chap-dtrace-intermediate-format.tex:
```diff
@@ -197,11 +241,13 @@ \subsection{Branch Format (B-Format)}
 It is used for a range of instructions including those to load values from
-integer and string constant tables, and to load and store variable values.
+integer and string constant tables, and to load and store variable values that
+are scalars.
```
, as well as those that store scalar values in variables.

gvnn3 commented on specification/instr/ldga.tex:
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
 \subsubsection*{Format}
 
-\textrm{LDGA \%rd, \%r1, \%r2}
+\textrm{LDGA \%rd, var, ndx}
```
We can afford the extra characters, s/ndx/index/

gvnn3 commented on specification/instr/ldga.tex:
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
-built-in variable based on the value in \registerop{r1} with an
-optional array index in \registerop{r2}.
+built-in variable based on the value in \registerop{var} with an
+optional array index in \registerop{ndx}. \\
```
Why the trailing \\ ?

dstolfa replied:
Makes the pdf look more legible IMO. Can remove if you think it shouldn't be there (same idea is in ldgs).

gvnn3 commented on specification/instr/ldga.tex:
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
+built-in variable based on the value in \registerop{var} with an
+optional array index in \registerop{ndx}. \\
+
+Contrary to  \instruction{ldgs}, the variable identifier is 8 bits long, and
```
s/Contrary to/Unlike the/

gvnn3 commented on specification/instr/ldgs.tex:
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
 
-The \instruction{ldgs} instruction looks up a user defined variable in
-a DIF program and places its value into \registerop{rd}.
+The \instruction{ldgs} instruction functions in two ways and is intended to be
```
s/functions in two ways/has two modes of operation/

gvnn3 commented on specification/instr/ldgs.tex:
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
-a DIF program and places its value into \registerop{rd}.
+The \instruction{ldgs} instruction functions in two ways and is intended to be
+used only for scalar values. The first mode of operation is when the value
+provided in \registerop{var} is lesser than DIF\_VAR\_OTHER\_UBASE. This will
```
s/lesser/less/

gvnn3 commented on specification/instr/ldgs.tex:
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
+in a DIF program. The result of this instruction will be put into the register
+\registerop{rd}. \\
+
+Contrary to the \instruction{ldga} instruction, the \registerop{var} field is
```
s/Contrary to/Unlike/

gvnn3 commented on specification/instr/stgs.tex:
```diff
@@ -18,17 +18,31 @@ \subsubsection*{Format}
-The \instruction{stgs} instruct stores a value from the
-\registerop{rd} register into a user defined variable.
+
+Much like \instruction{ldgs}, the instruction \instruction{stgs} operates
```
s/Much like/Similar to/

gvnn3 commented on specification/instr/stgs.tex:
```diff
@@ -18,17 +18,31 @@ \subsubsection*{Format}
 
-This instruction has no run-time failure modes beyond its constraints.
+This instruction will fail if the supplied value in the \registerop{var} field
+is lesser than DIF\_VAR\_OTHER\_UBASE.
```
s/lesser/less/

dstolfa commented:
I've addressed some of the issues you've brought up. If you believe that the `\\` should be removed, placing the text together, I'll update that as well.
